### PR TITLE
Enabling GPv2 in powershell

### PR DESCRIPTION
**Pull request by Aleksand4rZivanovic**
## Description

Added `IsGeneralPurposeV2` and `StorageIOps` parameters to `New-AzSqlInstance`, `Set-AzSqlInstance` to enable the creation of GPv2 instances.

Also edited some autocomplete options - removed gen4 and added gen8.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs [hunk 1]
```diff
@@ -284,6 +284,19 @@ public class SetAzureSqlManagedInstance : ManagedInstanceCmdletBase
         [PSArgumentCompleter("Regular", "Freemium")]
         public string PricingModel { get; set; }
 
+        /// <summary>
+        /// Gets or sets whether or not this is a GPv2 variant of General Purpose edition.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Whether or not this is a GPv2 variant of General Purpose edition.")]
+        public bool? IsGeneralPurposeV2 { get; set; }
```
Comment (YanaXu) [inline src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs:291]: better to add tests for added parameters and record the tests.

**Comment by YanaXu** [comment]
Hi @Aleksand4rZivanovic, please fix the CI errors. 
If the parameter name can't be updated, please suppress the issue.

FYI, the next major release code complete date is 05/13.

**Comment by NoriZC** [comment]
Hi @Aleksand4rZivanovic any updates? If this pr is targeting current release, please add test cases as suggested.

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**